File: src/main/java/my/study/repository/design/pattern/proxy/java/RealImp.java
package my.study.repository.design.pattern.proxy.java;

/**
 * @author djh on  2018/9/7 13:43
 * @E-Mail 1544579459@qq.com
 */
public class RealImp implements Subject {
    @Override
    public void someMethod() {
        System.out.println("我是实际实现类,即被代理类,这是业务方法!");
    }
}

File: src/main/java/my/study/repository/design/pattern/proxy/kotlin/RealImp.kt
package my.study.repository.design.pattern.proxy.kotlin

/**
 * @author djh on  2018/10/15 19:26
 * @E-Mail 1544579459@qq.com
 */
class RealImp : Subject {
    override fun someMethod() {
        println("这是真实实现的业务方法!")
    }
}